mapviz_plugins/src/marker_plugin.cpp

```diff
@@ -254,6 +286,53 @@ namespace mapviz_plugins
     }
   }
 
+  /**
+   * Gives a MarkerData that represents an arrow and a transform, this function
+   * will generate the points involved in drawing the arrow and then transform
+   * all of them into the target frame.
+   * @param[inout] markerData A marker that represents an arrow.
+   * @param[in] transform The tf that should be applied to the arrow's points.
+   */
+  void MarkerPlugin::transformArrow(MarkerData& markerData,
+                      const swri_transform_util::Transform& transform)
+  {
+    // The firstpoint in the markerData.points array always represents the
+    // base of the arrow.
+    StampedPoint& point = markerData.points.front();
+    tf::Point arrowOffset;
+    if (markerData.points.size() == 1)
+    {
+      // If the markerData only has a single point, that means its "point" is
+      // the base of the arrow and the arrow's angle and length are determined
+      // by the orientation and scale_x.
+      point.transformed_point = transform * (markerData.local_transform * point.point);
+      tf::Transform arrow_tf(tf::Transform(
+          transform.GetOrientation()) * point.orientation,
+          tf::Point(0.0, 0.0, 0.0));
+      point.transformed_arrow_point = point.transformed_point +
+          arrow_tf * point.arrow_point * markerData.scale_x;
+      arrowOffset = tf::Point(0.25, 0.0, 0.0);
+    }
+    else
+    {
+      // If the markerData has two points, that means that the start and end points
+      // of the arrow were explicitly specified in the original message, so the
+      // length and angle are determined by them.
+      point.transformed_point = transform * point.point;
+      point.transformed_arrow_point = transform * point.arrow_point;
+      // Also, in those mode, scale_y is the diameter of the arrow's head.
+      arrowOffset = tf::Point(0.25 * markerData.scale_y, 0.0, 0.0);
+    }
+
+    tfScalar angle = tf::Point(1.0, 0.0, 0.0).angle(point.transformed_arrow_point - point.transformed_point);
```

Comment:
Decided to use std::atan2 instead to get the angle of the line through those points.  It seems a little bit simpler.